beta-prompts/beta_prompts/templating_framework.py

```
@@ -0,0 +1,1029 @@
+        self.templates: dict[str, dict[str, Any]] = {}
+        self.metadata_index: dict[str, TemplateMetadata] = {}
+
+        # Load existing templates
+        self._load_registry()
+
+    def _load_registry(self):
+        """Load templates from registry directory"""
+
+        registry_file = self.registry_path / "registry.json"
+        if registry_file.exists():
+            with open(registry_file) as f:
```

Ruff (0.12.2)

`open()` should be replaced by `Path.open()`

(PTH123)